pypesto/sample/parallel_tempering.py

```diff
@@ -197,12 +198,27 @@ def compute_log_evidence(
                 f"Carefully check the results. Consider using beta_init='{BETA_DECAY}' for better results."
             )
 
-        burn_in = result.sample_result.burn_in
-        trace_loglike = (
-            result.sample_result.trace_neglogprior[::-1, burn_in:]
-            - result.sample_result.trace_neglogpost[::-1, burn_in:]
-        )
-        mean_loglike_per_beta = np.mean(trace_loglike, axis=1)
+        # compute burn in for all chains and then estimate mean of log likelihood for each beta
+        mean_loglike_per_beta = []
+        temps = []
+        for i_chain in reversed(range(len(self.betas))):
+            burn_in_i = geweke_test(result, chain_number=i_chain)
```

Comment:
Honestly, I am not really sure. In thermodynamic integration, usually the chains are regarded independently. The mixing properties for the parallel tempering algorithm still hold (here one does not check if all chains are converged, only if the first one is). Since we are interested in computing the expecation of the log likelihood per chain independently, I think computing the burn in independently is the right way to do it.

Reply:
Also not sure what would be the ideal way. But I think the mixing of an already converged chain with an unconverged chain is not a problem. Since we assess converge at the end, so after all sampling is done. So I would expect if the swap move does change things substantially that chains then are considered as unconverged anyways. Or at least the Geweke index increases beyond that point. 

So I don't see a problem with doing it per chain.

However, what happens to the tempered posterior if a chain does not show convergence at all and the burn-in is equal to the sample size?

Reply:
If a chain shows no convergence it is not used to compute the evidence

Reply:
> here one does not check if all chains are converged, only if the first one is

Is this saying that all chains are converged if the lowest temperature chain is converged? Then why not use the lowest temperature chain burn-in for all chains, when computing E[logL], too?

> Since we are interested in computing the expecation of the log likelihood per chain independently, I think computing the burn in independently is the right way to do it.

> if the swap move does change things substantially that chains then are considered as unconverged anyways

What I mean is that the Geweke test doesn't ensure convergence, it can say converged but just plotting the sample trace makes it clear it's not converged. So, even though this per-chain E[logL] is computed independently, I would guess you get some not-converged chains in there, just by applying the Geweke test multiple times (can plot all chains as a dumb check), in realistic problems I have seen a lot of false positives. So, checking for convergence "across all chains together", might be more accurate.

Reply:
> So, checking for convergence "across all chains together", might be more accurate for E[logL] computation.
What exactly do you mean by this? Then a multi-dimensional Geweke test across chains?

We can never be sure about convergence with this test, that is true.